compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -608,10 +608,19 @@ object SymDenotations {
          }
       )
 
-    /** Is this a denotation of a stable term (or an arbitrary type)? */
-    final def isStable(implicit ctx: Context): Boolean = {
+    /** Is this a denotation of a stable term (or an arbitrary type)?
+      * Terms are stable if they are idempotent (as in TreeInfo.Idempotent): that is, they always return the same value,
+      * if any.
```

Comment:
The relation between stable and idempotent was a bit surprising to me and @odersky, but I think I just convinced him (in a quick chat) this makes sense. Counterexamples welcome!